src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -196,13 +196,25 @@ internal static string ReadBufferAsString(ArraySegment<byte> buffer, HttpContent
             Encoding encoding = null;
             int bomLength = -1;
 
+            string charset = headers.ContentType?.CharSet;
+
             // If we do have encoding information in the 'Content-Type' header, use that information to convert
             // the content to a string.
-            if ((headers.ContentType != null) && (headers.ContentType.CharSet != null))
+            if (charset != null)
             {
                 try
                 {
-                    encoding = Encoding.GetEncoding(headers.ContentType.CharSet);
+                    // Remove at most a single set of quotes.
+                    if (charset.Length > 2 &&
+                        charset[0] == '\"' &&
+                        charset[charset.Length - 1] == '\"')
+                    {
+                        encoding = Encoding.GetEncoding(charset.Substring(1, charset.Length - 2));
```

Comment:
I assume that since this case would have previously thrown an exception, we expect it to be very rare, and thus the extra string allocation isn't particularly impactful?